File: code_examples/3-Structured_Problem_Solving/19-Loops_and_Ifs/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program is the Fizz Buzz program
 */

// process & output
for (let counter = 1; counter <= 100; counter++) {
  if (counter % 3 === 0 && counter % 5 === 0) {
      console.log("Fizz-Buzz");
  } else if (counter % 3 === 0) {
      console.log("Fizz");
  } else if (counter % 5 === 0) {
      console.log("Buzz");
  } else {
      console.log(counter);
  }
}

console.log("\nDone.")

File: code_examples/3-Structured_Problem_Solving/19-Loops_and_Ifs/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program is the Fizz Buzz program
 */

package main

import (
	"fmt"
)

func main() {
	// this function is the Fizz Buzz program

	// process & output
	for counter := 1; counter <= 100; counter++ {
		switch {
		case counter%3 == 0 && counter%5 == 0:
			fmt.Println("Fizz-Buzz")
		case counter%3 == 0:
			fmt.Println("Fizz")
		case counter%5 == 0:
			fmt.Println("Buzz")
		default:
			fmt.Println(counter)
		}
	}
	fmt.Println("\nDone.")
}
